src/docs/arch-checkpoint.dox

```diff
@@ -19,46 +21,71 @@ WT_CONNECTION::rollback_to_stable.
 
 # The checkpoint algorithm #
 
-A checkpoint can be broken up into 5 main stages:
+The checkpoint algorithm can be considered as a coordinator of multiple aspects of the system. It
+relies on the @ref arch-btree "btree subsystem" for traversing in memory pages, the @ref arch-block "block manager subsystem" for
+handling on-disk blocks and the @ref arch-metadata "metadata subsystem" for constructing and interpreting the metadata
+information.
 
-_The prepare stage:_
+A checkpoint can be broken up into the following stages:
+
+## Locking ##
+
+The first stage of checkpoint is to gather the necessary checkpoint locks. The most important being
+the \c checkpoint_lock. This lock ensures only one checkpoint takes place at time. More granular locks
+such as, dhandle, schema and flush locks are taken throughout the checkpoint process.
+
+## Eviction ##
+
+Before the checkpoint transaction begins (pinning content), the @ref arch-eviction "eviction server" is leveraged to
+reduce the dirty content in the cache. This has the benefit of utilizing the multi-threaded
+functionality of eviction and reduces the work required during the critical section of the
+checkpoint. The thread performing the checkpoint operation will wait for eviction to
+reach the target dirty content ratio unless no progress is made..
+
+## Prepare ##
 
 Checkpoint prepare sets up the checkpoint, it begins the checkpoint transaction, updates the global
 checkpoint state and gathers a list of handles to be checkpointed. A global schema lock wraps
 checkpoint prepare to avoid any tables being created or dropped during this phase, additionally the
 global transaction lock is taken during this process as it must modify the global transaction state,
 and to ensure the \c stable_timestamp doesn't move ahead of the snapshot taken by the checkpoint
-transaction. Each handle gathered refers to a specific b-tree. The set of b-trees gathered by the
-checkpoint varies based off configuration. Additionally clean b-trees, i.e. b-trees without any
-modifications are excluded from the list, with an exception for specific checkpoint configuration
-scenarios.
+transaction. The connection dhandle list is walked one handle at a time where each handle gathered
+refers to a specific b-tree. The set of b-trees gathered by the checkpoint varies based off
+configuration. Additionally clean b-trees, i.e. b-trees without any modifications are excluded from
+the list, with an exception for specific checkpoint configuration scenarios. See
+\ref skipping-checkpoints "Skipping checkpoints" for scenarios where b-trees are skipped during
+checkpoint. B-trees included in the checkpoint list then have all earlier checkpoints marked for
+deletion. Previous checkpoints can only be discarded if the associated blocks are no longer used.
```

Comment:
The sentence about all earlier checkpoints marked for deletion is not quite correct. The code might mark them for deletion and then later clears that flag when the checkpoint is required to be retained. If you keep the sentence then you should itemize or at least say what conditions retain checkpoints. Or instead of itemizing, you can change these sentences to something like "Earlier checkpoints are deleted when possible." or something high level like that.